llvm/lib/Target/AArch64/AArch64TargetTransformInfo.cpp

```diff
@@ -2131,6 +2135,25 @@ AArch64TTIImpl::instCombineIntrinsic(InstCombiner &IC,
   case Intrinsic::aarch64_sve_st4:
   case Intrinsic::aarch64_sve_st4q:
     return instCombineSVENoActiveUnaryErase(IC, II, 4);
+  case Intrinsic::aarch64_sve_cmpeq:
```

Comment:
As these integer comparison intrinsics compile to instructions that set the status register (even if the predicate is all inactive, see use of [PredTest](https://developer.arm.com/documentation/ddi0602/2024-06/Shared-Pseudocode/aarch64-functions-sve?lang=en#impl-aarch64.PredTest.3) in [CMP<cc>)](https://developer.arm.com/documentation/ddi0596/2020-12/SVE-Instructions/CMP-cc---wide-elements---Compare-vector-to-64-bit-wide-elements-), it might be worth referencing where the [ACLE](https://github.com/ARM-software/acle/blob/main/main/acle.md#programmers-model-of-global-state) declares this part of the global state as undefined?

Reply:
This is a good point. [AAPCS](https://github.com/ARM-software/abi-aa/blob/2982a9f3b512a5bfdc9e3fea5d3b298f9165c36b/aapcs64/aapcs64.rst) says this:

> The N, Z, C and V flags are undefined on entry to and return from a public interface.

I think intrinsic call is a public interface, but I am not sure. @paulwalker-arm could you help clarify this?

Reply:
The compare intrinsics (and C builtins) have no side effects and so the fact some of them may code generate to instructions that set the condition codes is not relevant.

From the ACLE's point of view the result of a compare builtin should be tested via the `svptest` builtins.  The compiler will see this sequence and decide whether it can use the fact a compare instruction also sets the condition codes to its advantage.